lib/pedicel-pay/helper.rb

```diff
@@ -10,10 +10,19 @@ def initialize(config: PedicelPay.config, pedicel_instance: nil)
 
     def self.ec_key_to_pkey_public_key(ec_key)
       # EC#public_key is not a PKey public key, but an EC point.
-      pub = OpenSSL::PKey::EC.new(ec_key.group)
-      pub.public_key = ec_key.is_a?(OpenSSL::PKey::PKey) ? ec_key.public_key : ec_key
+      group = ec_key.group
+      point = ec_key.is_a?(OpenSSL::PKey::PKey) ? ec_key.public_key : ec_key
+      asn1 = OpenSSL::ASN1::Sequence(
+        [
+          OpenSSL::ASN1::Sequence([
+            OpenSSL::ASN1::ObjectId('id-ecPublicKey'),
+            OpenSSL::ASN1::ObjectId(group.curve_name)
+          ]),
+          OpenSSL::ASN1::BitString(point.to_octet_string(:uncompressed))
+        ]
+      )
 
-      pub
+      OpenSSL::PKey::EC.new(asn1.to_der)
```

Comment:
Have you verified that this will work as intended on all Ruby versions that we want to support (2.7, 3.0, 3.1, 3.2)?

Reply:
Yes. Old version only worked on Ruby versions lower than 3.1 that didn't support OpenSSL >= 3.0. This code however works both on older and newer version of OpenSSL and Ruby.

Before I've changes this code, all runners in pedicel pipeline for Ruby 3.1 and Ruby 3.2 failed due to PKeys in OpenSSL 3.0 being immutable. After I updated it all runners are passing. Check the [runners results](https://github.com/cub8/pedicel/actions/runs/6012267367/job/16307290409).
These runners use following Ruby versions: 2.7.8, 3.0.6, 3.1.4 and 3.2.2.

Besides, my forks are used already in two apps and we didn't meet any problems yet on Ruby versions 3.0.6 and 3.2.0. The proper version of OpenSSL is used as well.